lib/at_cmd/at_cmd.c

```diff
@@ -162,133 +230,125 @@ static void socket_thread_fn(void *arg1, void *arg2, void *arg3)
 				"thread killed", errno);
 			close(common_socket_fd);
 			return;
-		} else if (item->data[bytes_read - 1] != '\0') {
+		} else if (buf[bytes_read - 1] != '\0') {
```

Comment:
What if `bytes_read` is 0?

Reply:
@evenl Can we read a 0-length from an AT socket?
Here we are using the common shared AT socket. If a user closes this socket, what will the return value be? Should we first check for `errno` or in case of error is the return value always -1?

Reply:
The errno is not valid without a -1 return value, otherwise you don't know which call left the errno. I haven't checked with the code yet if we can get a 0 here. I can try to do that later today

Reply:
As discussed offline, this is not a regression. However, if there is any real chance of this happening, I can rewrite the error handling part to reflect that (probably adding `bytes_read == 0` as a separate case).

Reply:
I have never seen it, and it doesn't make sense. But the AT interface in BSDLib allow for packages containing 0 bytes, it's not considered an error condition by BSDlib. So you can, in theory receive 0 bytes from the AT socket.

Reply:
Added handling for this case